src/config/resolver.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::VecDeque;
+
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+use url::Url;
+
+use crate::config::{is_default, Config, Source};
+
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum LinkType {
+  #[default]
+  Config,
+  GraphQL,
+  Protobuf,
+  Data,
+}
+
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct Link {
+  #[serde(default, skip_serializing_if = "is_default", rename = "type")]
+  pub type_of: LinkType, // Type of the link
+  #[serde(default, skip_serializing_if = "is_default")]
+  pub src: String, // Source URL for linked files
+  #[serde(default, skip_serializing_if = "is_default")]
+  pub id: Option<String>, // Id is used to refer at different places in the config
+  content: Option<String>, // Stores raw content
+}
+
+impl Link {
```

Comment:
I think ConfigReader can automatically resolve all links. We don't need `resolver`.

Reply:
even when type is different from Config (which will be implemented in the future)?

Reply:
I think publicly we should only expose one API either - reader or resolver.
I should be able to pass a configuration to one of the module that it should provide me with the final merged configuration with all the linked content inside it.